### PR TITLE
The slow march to tagging sql backend tests

### PR DESCRIPTION
**Pull request by benrudolph**
@dimagi/test-team Still need to figure out how to properly run these in travis, but this now splits out couch and sql tests so we can run them more easily. Should shave ~2 mins off the python sharded tests. Thank god for macros

cc: @kaapstorm

### CODE REVIEW

**Comment by snopoke** [comment]
nice, lovin it

**Comment by benrudolph** [comment]
(offlined with simon, for now not going to run with `--attr=sql_backend` until all modules have been properly tagged because it's impossible to run `--attr=sql_backend` with another module)

File: corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py [hunk 1]
```diff
@@ -69,6 +63,13 @@ def _assertListNotEqual(self, l1, l2, msg=None, include_ordering=True):
         else:
             self.fail(msg)
 
+
+class CouchCaseRebuildTest(TestCase, CaseRebuildTestMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super(CouchCaseRebuildTest, cls).setUpClass()
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: This method is just a pass-thru, right? Could be removed.

**Review by millerdev — APPROVED** [review]
I like this pattern a lot. When can we remove `run_with_all_backends`?

File: corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py (hunk 1 above)
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: ah yes, but i don't think it's supposed to be, i just didn't copy over `delete_all_cases`

**Comment by benrudolph** [comment]
hopefully soonish, there's _a lot_ of them out there

File: corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py [hunk 2]
```diff
@@ -24,23 +27,20 @@ def test_generate_xml_with_no_date_modified(self):
             CommCareCase(_id='test').to_xml(version)
 
 
-@override_settings(CASEXML_FORCE_DOMAIN_CHECK=False)
-class CaseBugTest(TestCase, TestFileMixin):
-    """
-    Tests bugs that come up in case processing
-    """
+class CaseBugTestCouch(TestCase, TestFileMixin):
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: nit: Saw another one of these that ends in `CouchOnly` I think that may be a better convention and easier to grep for

File: corehq/form_processor/tests/utils.py [hunk 3]
```diff
@@ -170,6 +172,10 @@ def _delete_all(db, viewname, **view_kwargs):
 )
 
 
+def sql_backend_case(cls):
```
Comment (emord) [inline corehq/form_processor/tests/utils.py]: why not use `test_sql_backend` or something similar? This name makes me think it's going to return a CommCareCase for a sql domain

**Review by emord — APPROVED** [review]

File: corehq/form_processor/tests/utils.py (hunk 3 above)
Comment (benrudolph) [inline corehq/form_processor/tests/utils.py]: true it used to be `sql_backend_test_case` but then nose kept picking it up as an actual test... so i dropped the test and now the name doesn't really make sense. open to ideas of something without test

File: corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py (hunk 2 above)
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: 👍

File: corehq/form_processor/tests/utils.py (hunk 3 above)
Comment (emord) [inline corehq/form_processor/tests/utils.py]: Hmm. yeah it looks like it picks up anything matching `(?:\b|_)[Tt]est`. Could either pass a specified matcher to nose or use something without test. maybe `ensure_sql_backend` although I'm not in love with that either.

Comment (benrudolph) [inline corehq/form_processor/tests/utils.py]: maybe just `use_sql_backend`?

Comment (emord) [inline corehq/form_processor/tests/utils.py]: :+1:

Comment (benrudolph) [inline corehq/form_processor/tests/utils.py]: https://github.com/dimagi/commcare-hq/pull/15024/commits/7f6ad7e40b1ab55a08c5a001e0621d9320eb7abd

File: corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py (hunk 2 above)
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: https://github.com/dimagi/commcare-hq/pull/15024/commits/1295de7d7ce8e55cbc5c299464208af361d5f5ad

**Review by emord — APPROVED** [review]

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 37 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 43 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py, line 336 - W391 blank line at end of file
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 47 - E302 expected 2 blank lines, found 1
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 261 - E501 line too long (116 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/tests/test_state_hash.py, line 107 - E303 too many blank lines (3)
* corehq/ex-submodules/casexml/apps/stock/tests/test_consumption_for_case.py, line 50 - W391 blank line at end of file

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 37 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 43 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py, line 336 - W391 blank line at end of file
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 47 - E302 expected 2 blank lines, found 1
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 261 - E501 line too long (116 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/tests/test_state_hash.py, line 107 - E303 too many blank lines (3)
* corehq/ex-submodules/casexml/apps/stock/tests/test_consumption_for_case.py, line 50 - W391 blank line at end of file

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 37 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 43 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py, line 336 - W391 blank line at end of file
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 47 - E302 expected 2 blank lines, found 1
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 261 - E501 line too long (116 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/tests/test_state_hash.py, line 107 - E303 too many blank lines (3)
* corehq/ex-submodules/casexml/apps/stock/tests/test_consumption_for_case.py, line 50 - W391 blank line at end of file

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 37 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py, line 43 - F821 undefined name 'CaseBugTestCouch'
* corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py, line 336 - W391 blank line at end of file
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 47 - E302 expected 2 blank lines, found 1
* corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py, line 261 - E501 line too long (116 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/tests/test_state_hash.py, line 107 - E303 too many blank lines (3)
* corehq/ex-submodules/casexml/apps/stock/tests/test_consumption_for_case.py, line 50 - W391 blank line at end of file

**Comment by snopoke** [comment]
Can you not mark it as `not_test`

**Comment by benrudolph** [comment]
i originally thought of that and then i thought wrapping a decorator that i use to wrap test classes with `nottest` would cause the test not to be run. admittedly i didn't actually test it out

**Comment by benrudolph** [comment]
✅

File: corehq/form_processor/tests/utils.py (hunk 3 above)
Comment (millerdev) [inline corehq/form_processor/tests/utils.py]: @benrudolph you can use [`nose.tools.nottest`](http://nose.readthedocs.io/en/latest/testing_tools.html#nose.tools.nottest) to decorate things that are not tests but nose thinks they are.

**Pull request merged**